src/clients/HubPoolClient.ts

```diff
@@ -165,74 +165,91 @@ export class HubPoolClient extends BaseAbstractClient {
   }
 
-  getDestinationTokenForDeposit(deposit: {
-    originChainId: number;
-    originToken: string;
-    destinationChainId: number;
-  }): string {
-    const l1Token = this.getL1TokenForDeposit(deposit);
-    const destinationToken = this.getDestinationTokenForL1Token(l1Token, deposit.destinationChainId);
-    if (!destinationToken) {
-      this.logger.error({
-        at: "HubPoolClient",
-        message: "No destination token found",
-        deposit,
-        notificationPath: "across-error",
-      });
+  // Returns the latest L2 token to use for an L1 token as of the input hub block.
+  protected getL2TokenForL1TokenAtBlock(
+    l1Token: string,
+    destinationChainId: number,
+    latestHubBlock = Number.MAX_SAFE_INTEGER
+  ): string {
+    if (!this.l1TokensToDestinationTokensWithBlock?.[l1Token]?.[destinationChainId]) {
+      throw new Error(`Could not find L2 token mapping for chain ${destinationChainId} and L1 token ${l1Token}`);
     }
-    return destinationToken;
-  }
-
-  getL1TokensToDestinationTokens(): L1TokensToDestinationTokens {
-    return this.l1TokensToDestinationTokens;
-  }
-
-  getL1TokenForDeposit(deposit: { originChainId: number; originToken: string }): string {
-    const l1Token = Object.keys(this.l1TokensToDestinationTokens).find((_l1Token) => {
-      return this.l1TokensToDestinationTokens[_l1Token][deposit.originChainId] === deposit.originToken;
-    });
-    if (!l1Token) {
+    // Find the last mapping published before the target block.
+    const l2Token: DestinationTokenWithBlock | undefined = sortEventsDescending(
+      this.l1TokensToDestinationTokensWithBlock[l1Token][destinationChainId]
+    ).find((mapping: DestinationTokenWithBlock) => mapping.blockNumber <= latestHubBlock);
+    if (!l2Token) {
       throw new Error(
-        `Could not find L1 Token for origin chain ${deposit.originChainId} and origin token ${deposit.originToken}!`
+        `Could not find L2 token mapping for chain ${destinationChainId} and L1 token ${l1Token} equal to or earlier than block ${latestHubBlock}!`
       );
     }
-    return l1Token;
-  }
-
-  getL1TokenCounterpartAtBlock(l2ChainId: number, l2Token: string, hubPoolBlock: number): string {
-    const l1Token = Object.keys(this.l1TokensToDestinationTokensWithBlock).find((_l1Token) => {
-      // If this token doesn't exist on this L2, return false.
-      if (this.l1TokensToDestinationTokensWithBlock[_l1Token][l2ChainId] === undefined) {
-        return false;
-      }
-
-      // Find the last mapping published before the target block.
-      return sortEventsDescending(this.l1TokensToDestinationTokensWithBlock[_l1Token][l2ChainId]).find(
-        (mapping: DestinationTokenWithBlock) => mapping.l2Token === l2Token && mapping.blockNumber <= hubPoolBlock
-      );
-    });
-    if (!l1Token) {
+    return l2Token.l2Token;
+  }
+
+  // Returns the latest L1 token to use for an L2 token as of the input hub block.
+  protected getL1TokenForL2TokenAtBlock(
```

Comment:
@nicholaspai Do you think this needs to be protected? `getXForDeposit()` is really convenient but there are also cases where we don't have a deposit object, and having to coerce the input to look like a deposit is inconvenient. Examples are when we have a Fill (no `originToken` or `quoteBlockNumber`), or in the InventoryClient where we are ignorant of Deposits entirely.

Any objections to removing `protected` here to allow these callers to use it?

Reply:
nope lets remove

Reply:
I have made this a public method and updated the mocks to implement the necessary instrumentation for mocking l1 -> l1 token mappings.